### DISCUSSION TITLE
Transfer deep learning model to FMU

### DISCUSSION THREAD

**Discussion by Alex-X-yang** [NTNU-IHB/PythonFMU#164] [Q&A]
Hello! @markaren 
I really appreciate your work in building an FMU from python code! That is quite helpful for me!
I have learned your example and successfully export ’SympySlave.fmu‘ and 'Resistor.fmu' and use it for simulation. 
I am trying to make my already trained deep learning model as an FMU and import it to other software. However, I can't correctly rewriter my code.
I am wondering if you could help me with this? Could I write my **simple** code below and take you several minutes to rewrite it to the right form and generate FMU? 
Thank you very much for your kind consideration and I am looking forward to your early reply :)

**Comment by markaren** [comment]
I can't help much. There are alot if things that can go wrong when importing into a tool like dymola. You don't say what failed or if you got an error. Numpy has caused issues in the past. Dymola might load Python on it's own which might cause issues. Especially on Linux.

Your code looks fine except for the fact that you load the network from file at each iteration, which I guess would lead to some performance issues. (You seem to use absolute path for the file which is fine if you run this on the same PC as you exported the model). You could also use lambdas to write into an array rather than declaring all these instance variables. 

I would start debugging in another tool to see if Dymola is the issue.

**Reply by Alex-X-yang** [reply]
The error write as: `Fatal:MWSolver caught an exception` means structurization exceptions

**Reply by Alex-X-yang** [reply]
I did use absolute path to load model

**Comment by Alex-X-yang** [comment]
After one day's hard work, I have solved most problems. Still one question hope you can answer.
I seem to understand what you mean load the network from file at each iteration, which lead low simulation speed. So, how to solve it? Do you have any suggestion?

**Reply by markaren** [reply]
The FMI standard defines a number of functions. step (invoked at each simulation step) is the only one you are forced to implemented by PythonFMU, but you are free to implement the rest.

e.g., https://github.com/NTNU-IHB/PythonFMU/blob/4524b3ea37de7cbe49490d1a0d694cd64e7b3665/pythonfmu/fmi2slave.py#L188

```python
def exit_initialization_mode(self):
  self. model = torch.load('filename.pth')
```

**Reply by Alex-X-yang** [reply]
Thank you very much!
